libraries/chain/resource_meter.cpp

```diff
@@ -59,26 +65,25 @@ uint64_t resource_meter::disk_storage_remaining()
       auto cost = _resource_limit_data.disk_storage_cost();
 
       if ( cost > 0 )
-         return session->remaining_rc() / cost;
-      else
-         return std::numeric_limits< uint64_t >::max();
+         return std::min( session->remaining_rc() / cost, _disk_storage_remaining );
    }
 
    return _disk_storage_remaining;
 }
 
-void resource_meter::use_network_bandwidth( uint64_t bytes )
+void resource_meter::use_network_bandwidth( int64_t bytes )
 {
    KOINOS_ASSERT( bytes <= _network_bandwidth_remaining, network_bandwidth_limit_exceeded, "network bandwidth limit exceeded" );
+   KOINOS_ASSERT( bytes >= 0, reversion_exception, "cannot consume negative network bandwidth" );
```

Comment:
Should this be a `network_bandwidth_limit_exceeded` exception type? See `use_compute_bandwidth`.